### PR TITLE
#23 - Make the cleaning script great again

### PR DESCRIPTION
**Pull request by Nabellaleen**
Build a select component querying the cleaning script flask server as suggested in https://github.com/arkhn/pyrog/issues/23

image

### CODE REVIEW

File: client/tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,10 @@
     "target": "es6",
     "allowJs": true,
     "lib": ["es6", "dom", "esnext"],
-    "baseUrl": "."
+    "baseUrl": ".",
+    "paths": {
+      "~/*": ["src/*"]
```
Comment (LaRiffle) [inline client/tsconfig.json]: What does does this one do?

File: client/src/components/selects/scriptSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  let items: IScript[] = [];
+  if (data && data.cleaningScripts) {
+    items = data.cleaningScripts.scripts;
+  }
+  return (
+    <ButtonGroup>
+      <TSelect<IScript>
+        disabled={false}
+        displayItem={(item: any) => {
+          return item.code || "None";
```
Comment (LaRiffle) [inline client/src/components/selects/scriptSelect.tsx]: Can item.code be empty?

File: client/src/components/selects/scriptSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  onClear
+}: IProps) => {
+  const { loading: queryLoading, error, data } = useQuery(cleaningScripts);
+  let items: IScript[] = [];
+  if (data && data.cleaningScripts) {
+    items = data.cleaningScripts.scripts;
+  }
+  return (
+    <ButtonGroup>
+      <TSelect<IScript>
```
Comment (LaRiffle) [inline client/src/components/selects/scriptSelect.tsx]: Stupid question, what is the semantic of `<TSelect<IScript> ... />` ?

File: client/tsconfig.json (hunk 1 above)
Comment (Nabellaleen) [reply client/tsconfig.json]: - [ ] Ajouter un commentaire

Comment (Nabellaleen) [reply client/tsconfig.json]: Thank you ! I've added a comment :)

File: client/src/components/selects/scriptSelect.tsx (hunk 3 above)
Comment (Nabellaleen) [reply client/src/components/selects/scriptSelect.tsx]: I'm not sure to understand what you're talking about semantic ;) Tell me if I don't answer to your question !

`TSelect<IScript>` means that we are implementing a `TSelect` taking as items only objects implementing the `IScript` interface.

Comment (LaRiffle) [reply client/src/components/selects/scriptSelect.tsx]: So it would be a `TSelect` but extended with attributes from `IScript`?

**Pull request merged**